### ISSUE TITLE
When strategy is 'free' include tasks inside roles are not executing

### ISSUE THREAD

**Issue by vinpk** [ansible/ansible#25807]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

strategy: free

##### ANSIBLE VERSION

```
ansible 2.3.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, Jan 19 2017, 14:48:08) [GCC 6.3.0 20170118]

```

##### CONFIGURATION

callback skippy used (tested without this for this bug - no change in behavior) 
##### OS / ENVIRONMENT

Linux/Ubuntu 
##### SUMMARY

When strategy free used dynamic include tasks inside a role is not executed and the play just finishes without executing the included tasks. 

##### STEPS TO REPRODUCE

main.yml
```
---
 - name: strategy_tester
   hosts: router1
   strategy: free
   gather_facts: no
   roles:
     - testra
```
roles/testra/tasks/main.yml
```
---
- name: Call OS specific config
  include: "{{inventory_hostname}}.yml"
```
roles/testra/tasks/router1.yml
```
---
- include: "justprint.yml"
```
roles/testra/tasks/justprint.yml
```
---
- debug:
    msg: "printed"
```

##### EXPECTED RESULTS

```
ansible-playbook test_strategy.yml

PLAY [tester] *****************************************************************************

TASK [testra : Call OS specific config] ***************************************************
included: /home/vinp/Desktop/ansible/my/tests/roles/testra/tasks/router1.yml for router1

TASK [testra : include] *******************************************************************
included: /home/vinp/Desktop/ansible/my/tests/roles/testra/tasks/justprint.yml for router1

TASK [testra : debug] *********************************************************************
ok: [router1] => {
    "msg": "printed"
}

PLAY RECAP ********************************************************************************
router1      : ok=3    changed=0    unreachable=0    failed=0   

```
##### ACTUAL RESULTS

```
ansible-playbook test_strategy.yml -vvvv
Using /etc/ansible/ansible.cfg as config file
Loading callback plugin skippy of type stdout, v2.0 from /usr/local/lib/python2.7/dist-packages/ansible/plugins/callback/__init__.pyc

PLAYBOOK: test_strategy.yml ***************************************************************
1 plays in test_strategy.yml

PLAY [tester] *****************************************************************************
META: ran handlers

TASK [testra : Call OS specific config] ***************************************************
task path: /home/vinp/Desktop/ansible/my/tests/roles/testra/tasks/main.yml:2
included: /home/vinp/Desktop/ansible/my/tests/roles/testra/tasks/router1.yml for router1

TASK [testra : include] *******************************************************************
task path: /home/vinp/Desktop/ansible/my/tests/roles/testra/tasks/router1.yml:2
included: /home/vinp/Desktop/ansible/my/tests/roles/testra/tasks/justprint.yml for router1
META: ran handlers
META: ran handlers

PLAY RECAP ********************************************************************************
router1      : ok=2    changed=0    unreachable=0    failed=0   

```

**Comment by s-hertel** [comment]
Hi @vinpk, does using include_tasks instead of include fix this?

I seem to be getting the result you want when I change testra/tasks/main.yml to:
```
---
- name: Call OS specific config
  include_tasks: "{{inventory_hostname}}.yml"
```
and change testra/tasks/router1.yml to:
```
---
- import_tasks: "justprint.yml"
```
The use of 'include' for tasks has been deprecated though I'm not really familiar with strategy so maybe this is a bug.

**Comment by vinpk** [comment]
Hi @s-hertel, I am not sure which version you used (or did you directly fork to the latest code 2.4?), When I tried (on Ansible versions 2.2.0, 2.3.1), got below errors for both 'include_tasks' and 'import_tasks'

```
ERROR! no action detected in task. This often indicates a misspelled module name, or incorrect module path.

The error appears to have been in '/home/vinp/Desktop/ansible/my/tests/roles/testra/tasks/main.yml': line 2, column 3, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

---
- name: Call OS specific config
  ^ here
```

I couldn't find any documentations also for these references..

**Comment by s-hertel** [comment]
Ah, okay! I am on the latest devel. I will look into this @vinpk, thanks for opening the issue.

**Comment by amanvidura** [comment]
@s-hertel  Has this issue been resolved?

**Comment by peterwillcn** [comment]
Use include_tasks fixed.

**Comment by ogmueller** [comment]
The issue seems to exists even in ansible v2.4.0.0.

All "include_tasks" are ignored when using strategy free. As soon as I change the strategy or remove it, it works as expected again.

**Comment by sivel** [comment]
This issue was resolved in ada84532710f4d2e717614b9f31a3ec24cf2239e and included in the 2.4.3 release.
